From c1bebe6d944ddf0accbc44ae31c18c6ccce42c72 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Fri, 08 Apr 2022 13:04:11 +0300
Subject: [PATCH] [SW-77251] gaudi3: update sram base address

For recovery firmware update operations, SRAM memory
is used for communication and fit copy from host/i2c.
In gaudi3, SRAM memory can be configured to run
in various mode. This patch add changes to set
SRAM address assuming SRAM is configured in mode 0 with
size of 48MB for given die.

Change-Id: I684881291990900b2bb9c2e2394abd3704ce969e
---

diff --git a/arch/gaudi3/arc/soc.h b/arch/gaudi3/arc/soc.h
index 4fa6a53..8bc9c50 100644
--- a/arch/gaudi3/arc/soc.h
+++ b/arch/gaudi3/arc/soc.h
@@ -44,6 +44,8 @@
 #define SZ_64K				0x00010000
 #define SZ_128K				0x00020000
 #define SZ_256K				0x00040000
+#define SZ_16M				0x01000000
+#define SZ_32M				0x02000000
 #define SZ_256MB			0x10000000
 /* 2MB for Gaudi2 . TODO: Confirm for gaudi3 */
 #define FLASH_VALID_MASK		0x1FFFFF
@@ -88,6 +90,9 @@
 #define HL_FWSRAM_OFFSET		0x3140000
 #define HL_PCIE_FWSRAM_OFFSET		0x7F00000
 #define SP_SRAM_BASE			(HL_GEN_MEM_BASE + HL_SCRATCHPAD_SRAM_OFFSET)
+#define HL_SRAM_MODE0_SIZE		(SZ_32M + SZ_16M) /* single die mode */
+#define HL_SRAM_MODE0_BASE		(HL_FW_MEM_BASE - HL_SRAM_MODE0_SIZE)
+#define SDRAM_BASE			HL_SRAM_MODE0_BASE
 
 #define HL_FW_MEM_BASE_FULL		0x0300007FD0000000ull
 #define HL_GEN_MEM_BASE_FULL		HL_FW_MEM_BASE_FULL
diff --git a/common/memory.h b/common/memory.h
index 29d28ae..00f69b9 100644
--- a/common/memory.h
+++ b/common/memory.h
@@ -7,9 +7,7 @@
 #define _MEMORY_H_
 
 #include <stdint.h>
-
-/* TODO: re-verify below offsets */
-#define SDRAM_BASE		0x10000000
+#include <soc.h>
 
 #define MEM_GP_BASE		(SDRAM_BASE + 0x40000)	/* SRAM_BASE + 256K */
 #define MEM_GP_SZ		0xC0000			/* 768K */
